### PR TITLE
Test for DecodeQR Module

### LINKED ISSUE #546: Tests for more modules; starting with QR code module jsqr
We are starting to grow this library substantially and could really use more tests for specific modules. One example is that this pull request is trying to update `jsqr` but we don't test that module, so we don't know if that update breaks the module: #532 

The test could look something like this one, and be in the same folder; we could use a data-url encoded QR code as test data, and confirm that we get the correct data out of the module!

Here's the module code itself:

https://github.com/publiclab/image-sequencer/tree/main/src/modules/DecodeQr

Once this is done, we could move on to each other module. This would be a great repeat first-timers-only issue!

****

This has been marked as a good candidate for becoming a [first-timers-only issue](http://www.firsttimersonly.com) like [these](https://code.publiclab.org/#r=all), meaning that it's simple, self-contained, and with some extra formatting, could be a great entry point for a new contributor. If you're familiar enough with this code, please consider [reformatting or reposting it as a first-timers-only issue](https://publiclab.org/notes/warren/10-31-2016/create-a-welcoming-first-timers-only-issue-to-invite-new-software-contributors), and then ping @publiclab/reviewers to get it labelled. Or, if this is not your first time, try to solve it yourself!

### PR DESCRIPTION
**Pull request by vishalbakshi**
Fixes #546

I have created a test for the DecodeQR Module in the test/modules/decodeqr.js file in my repo (vishalbakshi/image-sequencer > main)

This is my first time working with GitHub, as well as writing a test for an open source project so please let me know if I have not followed any common-sense conventions. 
-------

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

Thanks!

### CODE REVIEW

**Comment by jywarren** [comment]
This is great!!! I'd love to see it tested in the context of a sequencer "run" -- like we've done here:

https://github.com/publiclab/image-sequencer/blob/611016b022b22d1f89247e7b13f7650e1b65f2c7/test/modules/image-manip.js#L52-L63

Perhaps we could combine these lines with your standalone test file? Have you confirmed this is being run in our Travis test runner here on GitHub? 

Great work! Thank you!

**Comment by vishalbakshi** [comment]
Awesome---thanks!! 

The following Travis CI output was received after I opened the PR:

https://travis-ci.org/publiclab/image-sequencer/builds/473963530?utm_source=github_status&utm_medium=notification

Additionally, it shows "All checks have passed" for my "main" branch (screenshot below).

screen shot 2019-01-01 at 10 30 28 pm

Please let me know if there is something else with Travis that should pass. 

I will work on combining my test file with the structure of the image-manip.js file you have referenced--thanks!

File: test/modules/decodeqr.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+const decodeQR = require('../../src/modules/DecodeQR/Module.js')
```
[the remark this answers is not in the document]
Comment (vishalbakshi) [reply test/modules/decodeqr.js]: Thanks for catching that! I have found the typo in the relative path and have fixed it in my repo. I am still working on adding the tests to the image-manip.js file after which I will commit the update. I may post a couple questions here tomorrow if I can't get my tests to work today.

**Comment by jywarren** [comment]
Hmm, this is interesting! The tests seem thorough, but i see what you mean about them being kind of different than other tests. Why don't we ask someone from @publiclab/is-reviewers to read through this and offer their thoughts too? Thank you!

**Comment by HarshKhandeparkar** [comment]
Issue fixed

**Pull request closed without merging**